pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -275,6 +97,14 @@ func (g *Gatherer) GetName() string {
 // GetGatheringFunctions returns gathering functions that should be run considering the conditions
 // + the gathering function producing metadata for the conditional gatherer
 func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gatherers.GatheringClosure, error) {
+	newGatheringRules, err := g.fetchGatheringRulesFromServer(ctx)
+	if err != nil {
+		klog.Errorf("unable to fetch gathering rules from the server: %v", err)
+		klog.Info("trying to use cached gathering functions")
```

Comment:
Where are the gathering functions cached please? It seems that `g.gatheringRules` is an empty slice in case of an error and then the `gatheringFunctions` is an empty map or am I wrong?

Reply:
They are stored in `g.gatheringRules`. Yes, if there's no service on the very first run, it's gonna be empty. Cache works if the service disappears later.

Reply:
I see. OK

Reply:
Probably a detail, but when the `g.gatheringRules` are empty then the metadata is recorded with `null`. It would be better to put the error message there or something like that IMO.

Reply:
Because this will never work in disconnected clusters....